### PR TITLE
Add mouse support + some code cleaning + exit with ctrl+C

### PR DESCRIPTION
**Pull request by VincentFoulon80**
Here are some changes I made :
- Added mouse scroll to change page
- Added mouse click to select an element
- Refactored matches for resize and mouse_press events
- Added ctrl+C to exit the app

### CODE REVIEW

File: src/bop_interfaces/listbox.rs [hunk 1]
```diff
@@ -78,6 +78,10 @@ impl ListBox {
         pos + (self.page * self.screen.get_height() as usize)
     }
 
+    pub fn set_position(&mut self, pos: usize) {
```
Comment (LaineZ) [inline src/bop_interfaces/listbox.rs:81]: I didn't quite understand why this function method was needed. For API consistency? You can just ```listbox.position``` for getting or setting values

**Review by LaineZ — COMMENTED** [review]
Big thanks for your contribution! But there are moments I'd like to discuss.

File: src/bop_interfaces/listbox.rs (hunk 1 above)
Comment (VincentFoulon80) [reply src/bop_interfaces/listbox.rs:81]: Oh, I assumed that was a private property, since it's something managed internally by other methods

Comment (LaineZ) [reply src/bop_interfaces/listbox.rs:81]: Okay, i fix that

**Pull request merged**